fuzzamoto/src/targets/mod.rs

```diff
@@ -48,10 +48,48 @@ pub trait ConnectableTarget {
     fn is_connected_to<O: ConnectableTarget>(&self, other: &O) -> bool;
 }
 
-pub trait HasTipHash {
-    fn get_tip_hash(&self) -> Option<[u8; 32]>;
+pub trait HasGetBestBlockHash {
+    fn getbestblockhash(&self) -> Option<BlockHash>;
+}
+
+pub trait HasGetBlockCount {
+    fn getblockcount(&self) -> Option<u64>;
+}
+
+pub trait HasGetBlockHash {
+    fn getblockhash(&self, height: u64) -> Option<BlockHash>;
+}
+
+pub trait HasGetBlock {
+    fn getblock(&self, hash: BlockHash) -> Option<Block>;
+}
```

Comment:
These are used for the new generators to fetch the blocks that were accepted right?

I think we only need to be able to get a block by hash and we wont need the height to hash interface, because each block contains the hash of the previous block it is building on.